js/ui/status/payg.js

```diff
@@ -64,21 +70,35 @@ class PaygIndicator extends PanelMenu.SystemIndicator {
             REFRESH_TIME_SECS,
             () => this._timeoutRefresh());
         GLib.Source.set_name_by_id(this._timeoutRefreshId, '[gnome-shell] this._timeoutRefresh');
-    }
 
-    _timeoutRefresh() {
-        this._updateTimeRemainingAndSyncWhenReady();
-        return GLib.SOURCE_CONTINUE;
+        this._sessionModeUpdatedId = Main.sessionMode.connect('updated',
+            this._sync.bind(this));
+
+        this.connect('destroy', this._onDestroy.bind(this));
+
+        this._sync();
     }
 
     _onDestroy() {
-        super._onDestroy();
```

Comment:
You still need to preserve this chainup, don't you?

Reply:
Not really, this was actually wrong and this method never invoked as the base class doesn't define `_onDestroy()` or connects to the destroy signal.

Reply:
The base class being `PanelMenu.SystemIndicator` which inherits from `St.BoxLayout`.

Reply:
Okay